src/tvdb/client.py

```diff
@@ -0,0 +1,192 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Literal, overload, override
+
+import aiohttp
+from yarl import URL
+
+from src.settings import TVDB_API_KEY
+from src.tvdb.generated_models import (
+    MovieBaseRecord,
+    MovieExtendedRecord,
+    SearchResult,
+    SeriesBaseRecord,
+    SeriesExtendedRecord,
+)
+from src.tvdb.models import MovieResponse, SearchResponse, SeriesResponse
+from src.utils.log import get_logger
+
+log = get_logger(__name__)
+
+type JSON_DATA = dict[str, JSON_DATA] | list[JSON_DATA] | str | int | float | bool | None  # noice
+
+
+class _Media(ABC):
+    def __init__(self, client: "TvdbClient"):
+        self.client = client
+        self._id: int | None = None
+
+    @property
+    def id(self) -> int | str | None:
+        return self._id
+
+    @id.setter
+    def id(self, value: int | str) -> None:
+        self._id = int(str(value).split("-")[1])
```

Comment:
This looks wrong, the `value` param is annotated as `int | str`, if it's an int, you run `str()` on it, then split by `-` and take the second part? That will always produce an error, no ints contain `-`.

Reply:
`.split` won't cause any errors if `'-'` is not found. It is being ran on a string since str is used on `value`
```py
text = "Hello world"
result = text.split(",")
print(result)  # Output: ['Hello world']
```
The idea is to allow `["movie-12345", "12345", 12345]` to all be valid.

Reply:
![image](https://github.com/user-attachments/assets/844594ed-b92a-4f2b-839c-5ae67d2e97ee)

Reply:
If it's just for `movie-` or `series-`, I'd be much happier with `str.removeprefix` here